src/server/rpc/storage.rs

```diff
@@ -785,11 +786,9 @@ impl SetRange for PredicateBuilder {
     }
 }
 
-fn get_database_name(input: &impl GrpcInputs) -> Result<String, Status> {
-    Ok(org_and_bucket_to_database(
-        input.org_id()?,
-        &input.bucket_name()?,
-    ))
+fn get_database_name<'a>(input: &impl GrpcInputs) -> Result<DatabaseName<'a>, Status> {
```

Comment:
I think this actually wants to use `'static` too:

```suggestion
fn get_database_name(input: &impl GrpcInputs) -> Result<DatabaseName<'static>, Status> {
```

Otherwise the `'a` is saying the output lifetime is tied to something that comes out of nowhere (it's not tied to any of the input parameters). This compiles because the only lifetime that *could* satisfy this constraint, and the only lifetime that is actually used in practice, is `'static` (aka owned `String`s that don't have any references)

Reply:
Nice - that makes total sense.

Thanks!